vm-structure.md

```diff
@@ -1,7 +1,86 @@
-There is a standard template we use for our view models in both Java and Swift. This helps use not worry about stylistc 
-decisions and have a well-defined place to put protocols, methods, etc.
+There is a standard template we use for our view models in both Java and Swift. This helps us not worry about stylistc 
+decisions and have a well-defined place to put protocols, methods, interfaces, etc.
 
 ## Java
+```java
+// {List of imports}
+
+public interface MyViewModel {
+
+  interface Inputs {
+    /** {Alphabetized list of input functions with documentation} */
+    
+    /** Call to configure the view model with a project. */
+    void configureWith(Project project);
+
+    /** Call when the next page has been invoked. */
+    void nextPage();
+  }
+  
+  interface Outputs {
+    /** {Alphabetized list of output observables with documentation} */
+  
+    /** Emits the creator name to be displayed. */
+    Observable<String> creatorNameTextViewText();
+  }
+  
+  final class ViewModel extends ActivityViewModel<MyActivity> implements Inputs, Outputs {
+  
+    // {Constructor of the view model at the top.}
+    
+    public ViewModel(final @NonNull Environment environment) { 
+      super(environment);
+       
+      this.creatorNameTextViewText = this.project.map(p -> p.creator.name);
+    }
+    
+    // {Private helper methods (optional) before implementtion of interfaces.}
+    
+    private @NonNull String helper() {
```

Comment:
we should make all `private` helpers also `static` since they shouldn't depend on any instance variables, i.e. they should be passed all the data they need to do work.